packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -799,40 +799,45 @@ class FadeForwardsPageTransitionsBuilder extends PageTransitionsBuilder {
     Animation<double> secondaryAnimation,
     Color? backgroundColor,
     Widget? child,
-  ) => DualTransitionBuilder(
-    animation: ReverseAnimation(secondaryAnimation),
-    forwardBuilder: (BuildContext context, Animation<double> animation, Widget? child) {
-      return ColoredBox(
-        color:
-            animation.isAnimating
-                ? backgroundColor ?? Theme.of(context).colorScheme.surface
-                : Colors.transparent,
-        child: FadeTransition(
-          opacity: _fadeInTransition.animate(animation),
-          child: SlideTransition(
-            position: _secondaryForwardTranslationTween.animate(animation),
-            child: child,
-          ),
-        ),
-      );
-    },
-    reverseBuilder: (BuildContext context, Animation<double> animation, Widget? child) {
-      return ColoredBox(
-        color:
-            animation.isAnimating
-                ? backgroundColor ?? Theme.of(context).colorScheme.surface
-                : Colors.transparent,
-        child: FadeTransition(
-          opacity: _fadeOutTransition.animate(animation),
-          child: SlideTransition(
-            position: _secondaryBackwardTranslationTween.animate(animation),
-            child: child,
-          ),
-        ),
-      );
-    },
-    child: child,
-  );
+  ) {
+    final bool isOpaque = ModalRoute.of(context)?.opaque ?? false;
```

Comment:
Bit of an extreme nit, but this should probably default to true if `?.opaque` is null. I don't think it would actually matter, it should not be null for a ModalRoute, and routes that are not ModalRoutes currently do not use delegated transitions. But I think it's better if this defaults to true, as that's also ModalRoute's default.